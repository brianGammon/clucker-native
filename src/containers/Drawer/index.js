import React from 'react';
import { AsyncStorage, SafeAreaView, Text } from 'react-native';

export interface Props {
  navigate: any;
}
export default class Drawer extends React.Component<Props> {
  signOutAsync = async () => {
    await AsyncStorage.clear();
    const { navigation } = this.props;
    navigation.navigate('Auth');
  };

  render() {
    const { navigation } = this.props;
    return (
      <SafeAreaView>
        <Text onPress={() => navigation.navigate('Settings')}>Flock</Text>
        <Text onPress={this.signOutAsync}>Sign Out</Text>
      </SafeAreaView>
    );
  }
}
