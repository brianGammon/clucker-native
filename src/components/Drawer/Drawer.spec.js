import React from 'react';
import Drawer from '.';

describe('Drawer component:', () => {
  test('Should render', () => {
    const wrapper = shallow(<Drawer />);
    expect(wrapper).toMatchSnapshot();
  });
});
