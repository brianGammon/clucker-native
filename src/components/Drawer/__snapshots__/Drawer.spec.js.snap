// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer component: Should render 1`] = `
<SafeAreaView>
  <Component
    onPress={[Function]}
  >
    Flock
  </Component>
  <Component
    onPress={[Function]}
  >
    Sign Out
  </Component>
</SafeAreaView>
`;
